feature/607

AnimeOneLove2190 opened the pull request:
01
Added and fully equipped HallDataAccess, RowDataAccess, PlaceDataAccess, as well as the corresponding interfaces to them

AnimeOneLove2190 committed:
feature/607
01
Added and fully equipped HallDataAccess, RowDataAccess, PlaceDataAccess, as well as the corresponding interfaces to them

AnimeOneLove2190 committed:
feature/607/2
Add Controllers: HallController, RowController, PlaceController,
AddServices: HallCreate, RowCreate, PlaceCreate, HallUpdate(non-working)

artemlipinskiy commented on CinemaTicket.WebApi/Controllers/HallController.cs:
```diff
@@ -0,0 +1,34 @@
+        }
+        [HttpPost]
+        [Route("AddHall")]
+        public async Task CreateMovieAsync(HallCreate hallCreate)
```
CreateAsync или CreateHallAsync

artemlipinskiy commented on CinemaTicket.WebApi/Controllers/HallController.cs:
```diff
@@ -0,0 +1,34 @@
+        }
+        [HttpPost]
+        [Route("UpdateHall")]
+        public async Task UpdateMovieAsync(HallUpdate movieUpdate)
```
UpdateAsync или UpdateHallAsync

artemlipinskiy commented on CinemaTicket.WebApi/Controllers/PlaceController.cs:
```diff
@@ -0,0 +1,28 @@
+        }
+        [HttpPost]
+        [Route("AddPlace")]
+        public async Task CreateMovieAsync(PlaceCreate placeCreate)
```
CreateAsync или CreatePlaceAsync

artemlipinskiy commented on CinemaTicket.WebApi/Controllers/RowController.cs:
```diff
@@ -0,0 +1,28 @@
+        }
+        [HttpPost]
+        [Route("AddRow")]
+        public async Task CreateMovieAsync(RowCreate rowCreate)
```
CreateAsync или CreateRowAsync

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/HallService.cs:
```diff
@@ -0,0 +1,114 @@
+            if (hallFromDB.Rows != null || hallFromDB.Rows.Count > 0)
+            {
+                var rowsIds = hallFromDB.Rows.Select(x => x.Id).ToList();
+                var placesInHall = placeDataAccess.GetPlaceListAsync().Result.Where(x => rowsIds.Contains(x.RowId)).ToList();
```
await
Вообще можно из hallFromDb.Rows получить места через SelectMany

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/HallService.cs:
```diff
@@ -0,0 +1,114 @@
+                for (int i = 0; i < placesInHall.Count; i++)
+                {
+                    var soldTickets = placesInHall[i].Tickets.Where(x => x.IsSold == true).ToList();
+                    if (soldTickets.Count > 0)
```
Получается, что если есть проданные билеты, то нельзя обновлять зал?
Может лучше будет, если нельзя удалять ряды, в которых проданы билеты

AnimeOneLove2190 committed:
feature/607/3
Fixed protection of sold tickets and mechanisms for adding/deleting rows with the UpdateHallAsync method in HallService

AnimeOneLove2190 committed:
feature/607/4
Add RowUpdate Service

AnimeOneLove2190 committed:
feature/607/5
Added protection against duplicate numbers in RowCreate and RowUpdate Services

AnimeOneLove2190 committed:
feature/607/5
Add methods GetAsync, GetListAsync, DeleteAsync for HallService, RowService, PlaceService and GenreService,
Added corresponding controllers
